### PR TITLE
Add getAny to RichIterable

### LINKED ISSUE #548: Implement RichIterable.getAny().
In the Javadoc where `getFirst()` and `getLast()` are deprecated, there should be suggestion to use `getAny()`

### PR DESCRIPTION
**Pull request by alimcmaster1**
Closes #548 

Signed-off-by: alimcmaster1 <alimcmaster1@gmail.com>

### CODE REVIEW

**Comment by nikhilnanivadekar** [comment]
@alimcmaster1 thank you for your contribution.
There are couple of checkstyle failures: https://travis-ci.org/eclipse/eclipse-collections/jobs/414259660#L818
Can you please fix them before I can start the review? Thanks!

**Comment by alimcmaster1** [comment]
Thanks @nikhilnanivadekar,  CheckStyles failures are now resolved. Seems that there are some test failures after my second commit, which is strange as I only changed spacing. I can take a further look..

MemoryEfficientMutableSetParallelSetIterableTest>ParallelIterableTestCase.forEach_interruptedException:1025 Caught exception with cause of type<java.util.concurrent.ExecutionException>, expected cause of type <java.lang.InterruptedException>

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java [hunk 1]
```diff
@@ -104,6 +104,17 @@ default boolean notEmpty()
...
 
+    /**
+     * Returns any element of an iterable.
+     *
+     * @return an element of an iterable.
+     * @since 10.0
+     */
+    default T getAny()
+    {
+        return this.getFirst();
```
Comment (nikhilnanivadekar) [inline eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java]: I think @motlin was suggesting to delete to `this.getFirst()` only for OrderedIterable. For other types, it should be implemented.
@motlin correct me if I am wrong.

Comment (alimcmaster1) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java]: I understood the below comment to be as per my changes. But please let me know if ive misunderstood something here @motlin  

Motlin:
The fastest way to add getAny is as a default method on RichIterable that
delegates to getFirst but isn’t deprecated.

Comment (motlin) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java]: I think this is fine. IF we ever wanted to remove getFirst from RichIterable, but leave it on OrderedIterable, then we'd have to push this implementation down and implement getAny like getFirst in places like sets. But we probably won't remove methods.

File: unit-tests/src/test/java/org/eclipse/collections/impl/AbstractRichIterableTestCase.java [hunk 2]
```diff
@@ -1042,7 +1051,7 @@ public void sumDoubleConsistentRounding2()
 
         Assert.assertEquals(
                 33333.0,
-                this.newWith(list.toArray(new Integer[]{})).sumOfDouble(i -> 1.0d / 3.0d),
+                this.newWith(list.toArray(new Integer[] {})).sumOfDouble(i -> 1.0d / 3.0d),
```
Comment (motlin) [inline unit-tests/src/test/java/org/eclipse/collections/impl/AbstractRichIterableTestCase.java]: Are these whitespace changes causing CheckStyle failures?

Comment (alimcmaster1) [reply unit-tests/src/test/java/org/eclipse/collections/impl/AbstractRichIterableTestCase.java]: Adding them didn't seem to cause any CheckStyles failures on the build. My Intellij CheckSyles plugin seemed to add these... I have just revert the addition.

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java (hunk 1 above)
Comment (alimcmaster1) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java]: Thanks @motlin , any other changes required on this PR?

**Comment by motlin** [comment]
I just wanted to see what @nikhilnanivadekar thinks. Also if you can rebase/squash that would be great.

**Comment by nikhilnanivadekar** [comment]
LGTM. Please rebase/squash and this should be good to go.

**Comment by nikhilnanivadekar** [comment]
Can you please fix the commit message?
"Implement getAny() on RichIterable. Fixes #548"

**Comment by alimcmaster1** [comment]
Apologies @nikhilnanivadekar , ive fixed up the commit message now :)

**Comment by motlin** [comment]
It's squashed but not rebased. Shall I just "rebase and merge"?

**Comment by nikhilnanivadekar** [comment]
It messes up the network. Java 11 build is failing due to some issue in test utils, so we can ignore it for now

**Comment by nikhilnanivadekar** [comment]
@alimcmaster1 can you please rebase on to master? we will merge the PR. Thanks!

**Comment by alimcmaster1** [comment]
@nikhilnanivadekar this should now be rebased onto master. Thanks!

**Pull request merged**